### PR TITLE
Cherry-pick #264, #274, #295, #319

### PR DESCRIPTION
**Pull request by khieta**
## Description of changes

Added additional commits to `release/2.4.1` plus fixed the CHANGELOG.

The only changes that need manual review are the edits to the CHANGELOGs. All other commits are the result of clean a `cherry-pick`. The CHANGELOG edits need to be ported back to `main`; I will do this in a followup PR.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy/CHANGELOG.md [hunk 1]
```diff
@@ -42,65 +46,6 @@ validator will return an error for a policy that can never fire.
 
 ## 2.3.2
 
-### Removed
-- Move public API for partial evaluation behind experimental feature flag. To
-  continue using this feature you must enable the `partial-eval` feature flag.
-- Export the `cedar_policy_core::evaluator::{EvaluationError, EvaluationErrorKind}` and
```
Comment (khieta) [inline cedar-policy/CHANGELOG.md:48]: This large block of deleted text was duplicate from elsewhere in the file. Likely the result of some merging issue.

File: cedar-policy/CHANGELOG.md [hunk 2]
```diff
@@ -1,13 +1,17 @@
 # Changelog
 
 ## 2.4.1
-- Support `Request`s with `Unknown` fields for partial evaluation.
+### Added
+- New API to construct queries with `Unknown` fields for partial evaluation.
```
Comment (cdisselkoen) [inline cedar-policy/CHANGELOG.md:5]: we're adding a new API in a patch release?

**Review by cdisselkoen — APPROVED** [review]

File: cedar-policy/CHANGELOG.md (hunk 2 above)
Comment (khieta) [reply cedar-policy/CHANGELOG.md:5]: It's behind the experimental flag for partial evaluation -- should I clarify in the changelog?

Comment (andrewmwells-amazon) [reply cedar-policy/CHANGELOG.md:5]: Maybe something like "New experimental API..."

**Review by andrewmwells-amazon — APPROVED** [review]

**Review by shaobo-he-aws — APPROVED** [review]

**Pull request merged**